File: 135022193-main/2024/Done/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    int change;
    int quarters = 25;
    int dimes = 10;
    int nickels = 5;
    int pennies = 1;
    int coins = 0;

    do
    {
        change = get_int("change owed: ");
    }
    while (change < 0);

    if (change >= 25)
    {
        do
        {
            change = change - quarters;
            coins++;
        }
        while (change >= 25);
    }

    if (change >= 10)
    {
        do
        {
            change = change - dimes;
            coins++;
        }
        while (change >= 10);
    }

    if (change >= 5)
    {
        do
        {
            change = change - nickels;
            coins++;
        }
        while (change >= 5);
    }

    if (change >= 1)
    {
        do
        {
            change = change - pennies;
            coins++;
        }
        while (change >= 1);
    }

    printf("%d\n", coins);
}

File: 135022193-main/2024/Done/sentimental-cash/cash.py
from cs50 import get_float


def main():
    coins = 0

    change = int(get_change() * 100)
    coins = get_coins(change)
    print(coins)


def get_change():
    while True:
        change = get_float("Change: ")
        if change > 0:
            return change


def get_coins(change):
    coins = 0
    # quarters
    while change >= 25:
        change -= 25
        coins += 1
    # dimes
    while change >= 10:
        change -= 10
        coins += 1
    # nickels
    while change >= 5:
        change -= 5
        coins += 1
    # pennies
    while change >= 1:
        change -= 1
        coins += 1
    return coins


main()
